### PR TITLE
Added gmsh test.

### PR DESCRIPTION
**Pull request by odlomax**
Simple test to demonstrate single PE gmsh compatibility. Produces _xy_, _xyz_ and _lon-lat_ mesh outputs.

### CODE REVIEW

File: src/tests/grid/test_cubedsphere.cc [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    CASE("cubedsphere_grid_mesh_field_test") {
+
+      // THIS IS TEMPORARY!
+      // I expect this will be replaced by some more aggressive tests.
+
+      // Set grid.
+      const auto grid = atlas::Grid("CS-EA-24");
+
+      std::cout << grid->type() << std::endl;
+      std::cout << grid.size() << std::endl;
```
Comment (MarekWlasak) [inline src/tests/grid/test_cubedsphere.cc]: Replace cout

Comment (MarekWlasak) [reply src/tests/grid/test_cubedsphere.cc]: with  Log::info()

File: src/tests/grid/test_cubedsphere.cc [hunk 2]
```diff
@@ -0,0 +1,71 @@
+#include "atlas/functionspace/NodeColumns.h"
+#include "atlas/grid.h"
```
Comment (MarekWlasak) [inline src/tests/grid/test_cubedsphere.cc:10]: Remove Partitioner.h and add the text header with licence.

Comment (MarekWlasak) [reply src/tests/grid/test_cubedsphere.cc:10]: What about including 
```
#include "atlas/array/MakeView.h"
#include "atlas/field/Field.h"
```
You are using them, but they are hidden by auto

File: src/tests/grid/test_cubedsphere.cc (hunk 1 above)
Comment (odlomax) [reply src/tests/grid/test_cubedsphere.cc]: Done.

File: src/tests/grid/test_cubedsphere.cc (hunk 2 above)
Comment (odlomax) [reply src/tests/grid/test_cubedsphere.cc:10]: Done. Although it's reasonable to assume that `NodeColumns.h` will include the definition of a field, given it returns a non-pointer field.

**Review by MarekWlasak — APPROVED** [review]

**Pull request merged**